### PR TITLE
Annotating using Checker Framework, with Nullness Checkers

### PR DESCRIPTION
**Pull request by HuM4NoiD**

### CODE REVIEW

**Comment by HuM4NoiD** [comment]
Hi, I have created a branch `annotated` to try and find some bugs that may lead to NullPointerExceptions, The build may not work, as there are imports that are not included in maven.

File: src/main/java/org/apache/commons/cli/DefaultParser.java [hunk 1]
```diff
@@ -498,7 +510,7 @@ else if (matchingOpts.size() > 1 && !options.hasLongOption(opt))
             if (option.acceptsArg())
             {
                 handleOption(option);
-                currentOption.addValueForProcessing(value);
+                currentOption.addValueForProcessing(value);//TODO: Possible Dereference is currentOption is set to null
```
Comment (HuM4NoiD) [inline src/main/java/org/apache/commons/cli/DefaultParser.java]: I think I may have found a dereference here

File: src/main/java/org/apache/commons/cli/DefaultParser.java [hunk 2]
```diff
@@ -692,7 +706,7 @@ private void updateRequiredOptions(final Option option) throws AlreadySelectedEx
         {
             final OptionGroup group = options.getOptionGroup(option);
 
-            if (group.isRequired())
+            if (group.isRequired()) //TODO: Possible dereference if the option is not found
```
Comment (HuM4NoiD) [inline src/main/java/org/apache/commons/cli/DefaultParser.java]: And maybe another dereference here

**Comment by HuM4NoiD** [comment]
I was not able to find what may be the reason for the dereferences

**Comment by Kakarot-SSJ4** [comment]
The Travis build is failing due to a file with an unapproved license. You should look into it.
For integrating any work into the final product, it is important that the build succeeds.

**Comment by HuM4NoiD** [comment]
Travis build failed because I had not included a license in the `INSTRUCTIONS.md` and that only specifies what method I chose to detect and correct annotations for the NullnessChecker. If I include the license, diffs will be bloated and I was advised to keep the diffs as minimal as possible. Even If I do include the license I will get Symbol not resolved errors for the checker annotations and the build won't complete, as I chose the aliased `javacheck` command to detect errors, not the maven method. I faced problems with integrating Checker framework in maven, that's why I chose the above method.

**Comment by aherbert** [comment]
A license is required in `INSTRUCTIONS.md` so just copy it from README.md. The header policy applies to documentation files (see [Apache source headers](https://www.apache.org/legal/src-headers.html)).

**Comment by HuM4NoiD** [comment]
I am closing this as it is not relevant anymore

**Pull request closed without merging**